### DISCUSSION TITLE
pickMultipleObjects and GC

### DISCUSSION THREAD

**Discussion by reececasey** [visgl/deck.gl#6378] [Q&A] [answered]
Hi all, I am noticing that pickMultipleObjects is returning duplicates. The results seem to contain the same item with the layer status awaiting garbage collection. Even if I up the depth to 100, it will fill the array with the same point? 

Any ideas? This is happening on the GeoJsonLayer, I have seem some bugs raised around similar but nothing seems to come of it? 

Many thanks!

**Comment by Pessimistress** [comment, marked as answer]
Similar bugs have been reported before and fixed. If you are using the latest release please create a new bug report.

**Reply by reececasey** [reply]
There is one relating to MVT still open @Pessimistress We see it when using the MVT layer too. So I commented on here already. https://github.com/visgl/deck.gl/issues/6362 Sorry I didn’t mean multiple! I have seen bugs get fixed in older versions. Thats what I should have said.
It was more to see if it was approach based vs. a bug. 

I am on the latest version. Reason for not raising a bug is I was wondering if the react component re-renders more frequently could it be that the layers are not garbage collecting quickly enough? As some times it works other times the array is filled.

**Reply by Pessimistress** [reply]
When layer instances are recreated the older versions are discarded. That's what you saw as "awaiting GC". This is unrelated to the picking process.

#6362 is reported for MVT binary mode which is relatively new. MVTLayer also has its own picking handling logic so I didn't assume it was caused by a problem in the generic picking process. I will take a look.

**Reply by reececasey** [reply]
@Pessimistress I saw a comment in that issue that said it was happening regardless of the binary picking so wondered if it was a red herring. Happy to raise a separate bug though. Thank you for the support as always!

**Reply by Pessimistress** [reply]
I just did some quick testing and `pickMultipleObjects` seems to be working as expected. I'd say go ahead and create a new issue with a Codepen.

Edit: I can reproduce it now. Moving the discussion to #6362

**Reply by reececasey** [reply]
Thanks for digging into this @Pessimistress and raising a fix. Appreciated.
